### PR TITLE
BUFF-4218 Add .d.ts file to declare ableplayer a module

### PR DESCRIPTION
**Pull request by boronia**
To get rid of this typescript error:
```
$ tsc
src/web/containers/Block/VideoWidget/AblePlayer.tsx:3:28 - error TS7016: Could not find a declaration file for module 'ableplayer'. '/Users/von/projects/ableplayer/build/ableplayer.dist.js' implicitly has an 'any' type.
  Try `npm install @types/ableplayer` if it exists or add a new declaration (.d.ts) file containing `declare module 'ableplayer';`

3 import { AblePlayer } from 'ableplayer';
                             ~~~~~~~~~~~~
```

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -48,6 +48,13 @@ module.exports = function(grunt) {
                 dest: 'build/<%= pkg.name %>.js'
             },
         },
+        copy: {
+            files: {
+                cwd: '.',
+                src: 'scripts/ableplayer.d.ts',
+                dest: 'build/ableplayer.d.ts'
+            }
+        },
```
Comment (boronia) [inline Gruntfile.js:57]: I've put the file into `scripts` dir to be next to ableplayer-base.js, so had to have a grunt target to move it to `build` dir.

File: build/ableplayer.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+declare module 'ableplayer';
```
Comment (boronia) [inline build/ableplayer.d.ts]: I don't *have* to check it in, but so far all generated files are checked into the `build` dir, as per the original repo

**Review by ktml — APPROVED** [review]
Oh nice, so this is how it's done

File: scripts/ableplayer.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+declare module 'ableplayer';
```
Comment (hedgepigdaniel) [inline scripts/ableplayer.d.ts]: Surely you have to declare also an export `AblePlayer`? otherwise I think there will be a type error that the module ableplayer does hot define an export `AblePlayer`

Comment (boronia) [reply scripts/ableplayer.d.ts]: hmm so far there isn't a type error for AblePlayer 🤔 but yes now that you've mentioned I should include AblePlayer in here

Comment (hedgepigdaniel) [reply scripts/ableplayer.d.ts]: I mean a type error in OLC when you try to import it. You can probably test it with yarn link, and another option is using the @types/ directory in OLC (even just temporarily to ensure that the types work from the point of view of the user module)

Comment (boronia) [reply scripts/ableplayer.d.ts]: I know - I just tried it on Client (with yarn link) and there's no error.. I was also expecting it to but for some reason it did not complain about AblePlayer (maybe because VideoWidget is not `tsx`?) whoop VideoWidget is not the one that imports AblePlayer, it's AblePlayer.tsx... so yeh.. totally should be throwing an error there lol what's going on 😅

File: scripts/ableplayer.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+declare module 'ableplayer' {
+	export default AblePlayer;
```
Comment (boronia) [inline scripts/ableplayer.d.ts]: @hedgepigdaniel this ok?

Comment (boronia) [reply scripts/ableplayer.d.ts]: nope, that didn't work... let me try again lol

Comment (boronia) [reply scripts/ableplayer.d.ts]: Yeh nope, I tried sorry. Given that it's not actually causing any typescript errors if I leave it out, I'm going to leave it out.

File: scripts/ableplayer.d.ts (hunk 3 above)
Comment (boronia) [reply scripts/ableplayer.d.ts]: okok I'm not giving up yet... let me try one last thing...

Comment (boronia) [reply scripts/ableplayer.d.ts]: @hedgepigdaniel  BEHOLD!! GOT IT IN THE END! ☝️
Turns out, only export if it's from the tsx files, and `declare` if it's from an external non-tsx module. No need for `export` if I do `declare` but I *did* add one for the constructor 😁

Comment (boronia) [reply scripts/ableplayer.d.ts]: I think I earned the ✅ , give me a ✅ (🤞)

Comment (hedgepigdaniel) [reply scripts/ableplayer.d.ts]: Haha, yeah I haven't really worked out what the `declare module` thing is for - I think you can also skip that and have `export` statements - not really sure what the difference is

**Review by hedgepigdaniel — APPROVED** [review]
Well done :)
